operator/Dockerfile

```diff
@@ -57,20 +46,17 @@ COPY operator/build/status.sh status.sh
 RUN git init && git add .
 
 # Build the operator binary.
-RUN GOOS=$(go env GOOS) scripts/go-build.sh operator
+RUN GOOS=$(go env GOOS) CGO_ENABLED=0 scripts/go-build.sh operator
```

Comment:
Disabling CGO looks rather arbitrary. This will create a mismatch between upstream (this) and downstream builds where CGO is not disabled. Having such mismatch can lead to different behavior which isn't good because downstream images are not as thoroughly tested end-to-end as in upstream CI.

Last time I checked with Malte, there were specific reasons why CGO **was** enabled. I don't want to be a broken telephone relaying that info, particularly because I will have to research it myself. I believe there's more than "just" rocksdb, there are some tracing concerns that customers have.

Could you please search around for considerations why `main` and other images are built with CGO?

Reply:
In main we need rocksb to be linked. Here we don't have any external C libs so we don't need CGO.
The only one drawback I found is when disabling CGO we end up using go network stack and DNS may behave differently.

Reply:
Thanks Tomek but I don't think this info is sufficient. Have you read what I wrote just above?

> Last time I checked with Malte, there were specific reasons why CGO **was** enabled. I don't want to be a broken telephone relaying that info, particularly because I will have to research it myself. I believe there's more than "just" rocksdb, there are some tracing concerns that customers have.

Re-stating the same thing. CGO allows customers better trace the code execution because they can attach certain tools to binaries when they are built with CGO. This is real requirement from real StackRox customers.
I suggest you to research more on this topic. Either of two outcomes are possible:

* This is still the requirement and we still have to do it for the operator.
* This is not a requirement any more and we can stop doing that.

Reply:
Got it. I'll find answer about this requirement. For now... how about ubi8-micro?

Reply:
Thanks Tomek, sounds good.